Update sqlitebrowser with new variant +sqlcipher, and new Portfile for sqlcipher

z7z8th opened the pull request:
#### Description

<!-- Note: it is best to make pull requests from a branch rather than from master -->

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'printf "%s\n" "macOS `sw_vers -productVersion` `sw_vers -buildVersion` `uname -m`" "`xcodebuild -version|awk '\''NR==1{x=$0}END{print x" "$NF}'\''`"'|tee /dev/tty|pbcopy
-->
macOS 11.6.1 20G224 x86_64
Xcode 13.2.1 13C100

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vs install`?
- [ ] tried a full install with `sudo port -vst install`? *Can't build in trace mode. master branch version can't either.* 
- [x] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

i0ntempest commented:
Since the project is hosted on GitHub the github portgroup will make stuff easier, it will set `master_sites`, `distname`, and `version` for you. See https://guide.macports.org/chunked/reference.portgroup.html for a guide, and see https://github.com/macports/macports-ports/blob/master/net/qBittorrent/Portfile for an example

reneeotten commented on databases/sqlitebrowser/Portfile:
```diff
@@ -60,15 +62,29 @@ platform darwin {
 
-    use_parallel_build  no
-    depends_lib-append  port:sqlite3
+    use_parallel_build  yes
```
was the [original issue](https://trac.macports.org/ticket/60626) for why this was added resolved?

reneeotten commented on databases/sqlcipher/Portfile:
```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           clang_dependency 1.0
```
it's a bit surprising to me that this is needed as it implies that the port is needed for building recent clangs (at least if I read the PG correctly). Given that this port wasn't present before and clangs build fine I am a bit confused about this, but must admit that I haven't spend much (or any) time on this

z7z8th replied:
This portfile is copied from sqlite3, will delete this line.

reneeotten commented on databases/sqlcipher/Portfile:
```diff
@@ -0,0 +1,137 @@
+platforms           darwin
+license             public-domain
+
+maintainers         { gmail.com:ybzhao1989 @z7z8th } openmaintainer
```
no need for spaces around the `{`

reneeotten commented on databases/sqlcipher/Portfile:
```diff
@@ -0,0 +1,137 @@
+                    an example of how to use the SQLite3 library.
+
+homepage            https://www.zetetic.net/sqlcipher/
+master_sites        https://github.com/sqlcipher/sqlcipher/archive/refs/tags/
```
use github PG instead

reneeotten commented on databases/sqlcipher/Portfile:
```diff
@@ -0,0 +1,137 @@
+
+# livecheck.type      regex
+# livecheck.url       https://www.zetetic.net/sqlcipher/verify/
+# livecheck.regex     sqlcipher-(4(?:\\.\[0-9\]+)+).zip
```
please don't add commented out lines; switching to the GitHub PG will set this automatically anyway so it can be removed

reneeotten commented on databases/sqlcipher/Portfile:
```diff
@@ -0,0 +1,137 @@
+#                     CFLAGS="-DSQLITE_HAS_CODEC"
+
+#     configure.cppflags-prepend  -I${worksrcpath}
+# }
```
same as above, please no commented out code

reneeotten commented on databases/sqlcipher/Portfile:
```diff
@@ -0,0 +1,137 @@
+                              sha256  87458e0e16594b3ba6c7a1f046bc1ba783d002d35e0e7b61bb6b7bb862f362a7 \
+                              size    17611236
+
+    use_zip                   no
```
this is the default so no need to add it

reneeotten commented on databases/sqlcipher/Portfile:
```diff
@@ -0,0 +1,137 @@
+    build.target              sqldiff sqlite3_analyzer
+    destroot {
+        xinstall -m 755 ${worksrcpath}/sqldiff ${destroot}${prefix}/bin
+        xinstall -m 755 ${worksrcpath}/sqlite3_analyzer ${destroot}${prefix}/bin
```
please use `0755`

z7z8th commented:
Thanks @i0ntempest @reneeotten for the review. Fixed them in the new commits.
I also changed to license of sqlcipher, since it's actually BSD.

reneeotten commented on databases/sqlcipher/Portfile:
```diff
@@ -0,0 +1,114 @@
+PortGroup           github 1.0
+
+name                sqlcipher
+# don't forget to update the checksums for sqlite3-tools when updating sqlite3
```
this comment seems copy-paste as well... please update accordingly

reneeotten commented on databases/sqlcipher/Portfile:
```diff
@@ -0,0 +1,114 @@
+
+homepage            https://www.zetetic.net/sqlcipher/
+github.setup        sqlcipher sqlcipher 4.4.2 v
+
```
please add `github.tarball_from archive`  or `release` and that will require to update the `checksums`

z7z8th replied:
added `github.tarball_from archive` since no release in sqlcipher repo.
also added openssl11 as dependency to fix the workflow checks.

z7z8th commented:
@reneeotten Sorry, the dependency is not openssl11, should be openssl. Can you trigger the lint & build check again? Thanks.

i0ntempest commented:
> @reneeotten Sorry, the dependency is not openssl11, should be openssl. Can you trigger the lint & build check again? Thanks.

Would be better to use the openssl portgroup.

z7z8th commented:
@i0ntempest Thanks, I can try that, shoud I add OpenSSLException to the license part if it works?

z7z8th commented:
openssl portgroup is used, but still using openssl 3, since openssl 3 is linked to qt5-qtbase, and sqlitebrowser depends on qt5-qtbase. hence openssl 3 works fine with sqlcipher.

i0ntempest commented:
Good, just a note `openssl.branch` defaults to 3 so you don’t have to specify it. And if you’re done with editing, squash your changes and I’ll let workflows run on this PR.

z7z8th commented:
@i0ntempest Removed openssl.branch to use the default value, squashed and force pushed.

i0ntempest commented:
You squashed them all into one... ideally it should be one commit per portfile.

z7z8th commented:
sorry, split to 2 commits.

i0ntempest commented:
Right, checks are fine. If theres nothing from other members in 24 hours I’ll merge this.

reneeotten reviewed (changes requested):
@z7z8th @i0ntempest this doesn't seem to be a very well vetted portfile / PR yet. Please do make sure that all configure arguments/cppflags and such apply to this port; I've left a few comments of things that stood out directly by looking at the CI. Once you guys have confirmed everything is correct locally and updated this PR feel free to merge. I have no interest / knowledge about this port and leave it up to you to finish.

z7z8th commented:
Thanks @reneeotten @i0ntempest, I removed several obsolete flags from sqlcipher, -DSQLITE_HAS_CODEC is the flag to enable those security enhancements and should be kept. 
And depends_libs is also fixed.

z7z8th commented:
Thanks @reneeotten @i0ntempest, I removed several obsolete flags from sqlcipher, -DSQLITE_HAS_CODEC is the flag to enable those security enhancements and should be kept. 
And depends_libs is also fixed.

i0ntempest commented:
Found in the sqlcipher configure log:
```
  configure: WARNING: *** Without Tcl the regression tests cannot be executed ***
  configure: WARNING: *** Consider using --with-tcl=... to define location of Tcl ***
```
If you're interested in adding tcl support please do so, otherwise let me know and I'll merge it without support for tcl.

z7z8th commented:
Got, the sqlite3 default port didn't include tcl too, sqlite3 has a subport named sqlite3-tcl to support tcl which requires another archive of source code which sqlcipher does have. Hence no test is defined in portfile, so I think it's ok to go on without tcl support.
There's one conversation I can't resolve due to the commits not exist anymore, must be caused by the force pushes.

reneeotten reviewed (dismissed):
You add a default variant in the "sqlitebrowser" port, which warrants an increase in "revision"

z7z8th commented:
@reneeotten Thanks, the revision is updated now.

i0ntempest commented:
The port tries to install stuff into /S/L/F on 10.12 and older: https://build.macports.org/builders/ports-10.12_x86_64-builder/builds/177011/steps/install-port/logs/stdio

z7z8th commented:
I don't have a 10.12 macOS, will try to fix this by disable tcl. @i0ntempest
